docs/pages/plugins-manifest.md

```diff
@@ -81,55 +81,78 @@ Defines the minimum version of Sketch required to run the plugin. This string mu
+| `handlers`    | If more fine-grain control is required use the `handlers` field instead of `handler`, see below for in-depth documentation                                                                                                                                                                                                                                                                                       |
 
 ##### `handlers`
 
-An object used to specify the mapping between events (like Actions, calling the command, etc.) and function names.
+An object used to fully specify a command's _setup > run > tearDown_ lifecycle, as well as work with [Actions](/guides/action-api/). When specified the `handler` field is ignored.
 
-There are 4 different keys:
+The `handlers` object has four fields:
 
-- `run` which is the name of the function that should be called when the command is run (so it is the same as the `handler` key which is ignored when `handlers` is specified).
-- `setUp` which is the name of the function to call before the command is called (not to be confused with the `Startup` action).
-- `tearDown` which is the name of the function to call after the command has finished to run (not to be confused with the `Shutdown` action).
-- `actions` which is an object where each property name should be the name of an `Action` while the value should be the name of the function to call. (see [the Action guide](/guides/action-api/) for more information)
+- `run` The function that should be called when the command is run (this is conceptually the same as defining `handler`)
+- `setUp` The function to call before the command is called (not to be confused with the `Startup` action)
+- `tearDown` The function to call after the command has finished to run (not to be confused with the `Shutdown` action)
+- `actions` See the [Actions](/guides/action-api/) page for an in-depth guide
+
+For example, the following `commands` definition maps the `SelectionChanged.finish` action to the `onSelectionChanged` JS function,
 
 ```json
-"handlers": {
-  "actions": {
-    "SelectionChanged.finish": "onSelectionChanged"
-  }
+{
+  "commands": [
+    {
+      "name": "Selection Changed",
+      "identifier": "selection-changed",
+      "script": "./selection-changed.js",
+      "handlers": {
+        "actions": {
+          "SelectionChanged.finish": "onSelectionChanged"
+        }
+      }
+    }
+  ]
 }
 ```
 
+The function in `selection-changed.js` would be written as,
+
 ```js
 // onSelection handler
-export function onSelection(context) {
-  var doc = context.document;
-  var selection = context.selection;
-  …
+function onSelection(context) {
```

Comment:
so I think we need to be consistent in the docs. All the introduction is based on skpm because it's a lot easier to setup. So either we consider skpm the default, and the examples should use `export function`, or we need to rewrite the introduction

Reply:
This is not quite true. When I rewrote the documentation I assumed vanilla implementation to be default and when `skpm` is mentioned, it's specifically a provided as a convenience.

Reply:
True, but in the guides, skpm is always the first option (for "Create a plugin", "Publish a plugin", "update a plugin", etc.)

Reply:
That's right, but `skpm` is specifically mentioned in these guides. It might be a good reason to add a litte note in the _Create a plugin_ guide making it obvious because that's the only place where a handler is mentioned.

Reply:
Agreed consistency is key. As an end-user to the docs it wasn't clear to me which one was the default usage pattern. If I just arrive on the manifest.json page for example, I have no idea whether skpm is the first option on other pages or not, so that's not helpful. It's awkward that there are two competing approaches to creating a plugin, and while that's the case I think the documentation policy _has_ to always make it abundantly explicit which one is being discussed at every point.